### ISSUE TITLE
Move from es5 to es6 to support ENUMs and MAPs.

### ISSUE THREAD

**Issue by martinfrances107** [jsdrupal/drupal-admin-ui#552]
We have a series of micro issue all related to getting typescript working. Which I am going to stitch into a story before I point out the next problem.

Here is a description of what we have so far.

#533 Make MessageSeverity constants a ENUM

is trying to convert a small aspect of the design to typescript 

BUT 

#548 removes the unmet dependency issue which were getting in the way for me actually seeing the problem.
#539 shows that we don't actually feed typescript files into our linter - also prettier ignores them.

Now that we have a solution for #539  under consideration ( PR #551 )

I have returned to the ENUM problem and I am seeing error messages that "ENUM is a reserved word."

Googling around the internet  for typescript and create-react app suggests that this is a problem with the babel parser.

So in short this issue aims to fix that... possibly but updating/changing the parser.

**Comment by martinfrances107** [comment]
The next  PR does three simple things 

A) yarn remove react-scripts 
B) yarn add react-scripts-ts 
C) adjusted nightwatch.conf.js

**Comment by martinfrances107** [comment]
Above when I said 

>So in short this issue aims to fix that... possibly but updating/changing the parser.

The best way to resolve this and start supporting developments to the language like ENUM 
is to move from es5 to es6 

Here is a quote from the eslint documentation showing that the current default is es5

https://eslint.org/docs/2.0.0/user-guide/configuring

> ecmaVersion - set to 3, 5 (default), or 6 to specify the version of ECMAScript you want to use.

PR is comming.

**Comment by martinfrances107** [comment]
I have been thinking about this PR... and I can see a mistake I made.

I think maybe two PR should be merged into one...

A) #533 has outstanding issues involving ENUMS which are solved by this PR.

similarly 

B) This issue (#552)  needs an example of ENUM and MAP to verify PR is a good idea.

If a reviewer wants it... I will smoosh these two together... I will just saying...

**Comment by martinfrances107** [comment]
Thank you. ❤️
